Core/Converters/JsonLeakySortedDictionaryConverter.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+
+using log4net;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace CKAN
+{
+    /// <summary>
+    /// [De]serializes a dictionary that might have some questionably
+    /// valid data in it.
+    /// If exceptions are thrown for any key/value pair, leave it out.
+    /// Removes CkanModule objects from AvailableModule.module_version
+    /// if License throws BadMetadataKraken.
+    /// </summary>
+    public class JsonLeakySortedDictionaryConverter<K, V> : JsonConverter
+    {
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            var dict = new SortedDictionary<K, V>();
+            foreach (var kvp in JObject.Load(reader))
+            {
+                try
+                {
+                    dict.Add(
+                        (K)Activator.CreateInstance(typeof(K), kvp.Key),
+                        kvp.Value.ToObject<V>());
+                }
+                catch (Exception exc)
+                {
+                    log.WarnFormat($"Failed to deserialize {kvp.Key}", exc);
```

Comment:
`kvp.Key` is the module version, so we get warning messages like this one, which isn't terribly useful for debugging:
> 12560 [1] WARN CKAN.JsonLeakySortedDictionaryConverter`2[[CKAN.Versioning.ModuleVersion, ckan, Version=1.30.0.0, Culture=neutral, PublicKeyToken=null],[CKAN.CkanModule, ckan, Version=1.30.0.0, Culture=neutral, PublicKeyToken=null]] (null) - Failed to deserialize 1

Ideally we could print the identifier as well. If I've read the [docs](https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_JsonConverterAttribute.htm) correctly, it should be possible to pass custom parameters for a custom constructor in the attribute: 
```csharp
[JsonConverter(typeof(JsonLeakySortedDictionaryConverter<ModuleVersion, CkanModule>),this.identifier)]
internal SortedDictionary<ModuleVersion, CkanModule> module_version = ...
```
I haven't tried this out yet, though.

Reply:
I feared it wouldn't be that easy 😢
Then we'll just have to live with it.

Reply:
Hmm, if we logged some or all of `kvp.Value.ToString()`, I think that should give us the JSON of the module that failed to deserialize. Is that better than just the key?

Reply:
That's actually a good idea, this way we also have the "faulty" module right away without having to ask for a upload of the registry. (Assuming we get the logs, of course)

Reply:
OK, just force-pushed that. Also I should have been using `Warn` instead of `WarnFormat` all along because there were no format parameters and I wanted the special exception handling.